index.js

```diff
@@ -152,31 +178,49 @@ module.exports = (bookshelf, options) => {
      * @param {Boolean} patch If that operation was executed with a patch
      * @param {String} operation The name of the operation performed previously
      * from the backup
+     * @param {Object} historyOptions The history options of the model
      * @param {Object} [transacting] A valid transaction object
-     * @return {Number} The number of rows affected by the restoration
      */
-    history (model, patch, operation, transacting) {
-      let fields = model.historyOptions.fields
-      let execute = transacting => {
-        let forge = {}
+    history (model, patch, operation, historyOptions, transacting) {
+      const fields = model.historyOptions.fields
+      const execute = transacting => {
+        const forge = {}
         forge[fields.resource_id] = model.id
         forge[fields.resource_type] = model.tableName
 
+        if (!transacting) {
+          console.warn('No transaction detected at time of History write. In the future, Bookshelf-History will enforce a transaction.')
+        }
+
         return model.historyOptions.model.forge(forge)
-        .fetch({ transacting, require: false })
-        .then(row => row ? Number(row.get(fields.sequence)) + 1 : 1)
-        .then(sequence => {
-          let data = {}
-          data[fields.sequence] = sequence
-          data[fields.resource_id] = model.id
-          data[fields.resource_type] = model.tableName
-          data[fields.data] = JSON.stringify(model)
-          data[fields.patch] = Boolean(patch)
-          data[fields.operation] = operation
-
-          return model.historyOptions.model.forge(data)
-          .save(null, { transacting })
-        })
+          .fetch({ transacting, require: false })
+          .then(row => row ? Number(row.get(fields.sequence)) + 1 : 1)
+          .then(sequence => {
+            const data = {}
+            data[fields.sequence] = sequence
+            data[fields.resource_id] = model.id
+            data[fields.resource_type] = model.tableName
+            data[fields.data] = JSON.stringify(model.attributes)
+            data[fields.patch] = Boolean(patch)
+            data[fields.operation] = operation
+
+            if (historyOptions.authorCallback && typeof historyOptions.authorCallback === 'function') {
+              const metadata = historyOptions.authorCallback(model)
```

Comment:
Not sure I love the naming of the function "authorCallback".  usually you start with a verb so I'm thinking of authoring.  Maybe something like "getAuthorMetadata" or something